splice_machine/src/main/java/com/splicemachine/derby/stream/function/SetCurrentLocatedRowAndRowKeyFunction.java

```diff
@@ -17,27 +17,31 @@
 import com.splicemachine.db.iapi.sql.execute.ExecRow;
 import com.splicemachine.db.iapi.types.HBaseRowLocation;
 import com.splicemachine.derby.iapi.sql.execute.SpliceOperation;
+import com.splicemachine.derby.impl.sql.execute.operations.ProjectRestrictOperation;
+import com.splicemachine.derby.impl.sql.execute.operations.ScrollInsensitiveOperation;
+import com.splicemachine.derby.impl.sql.execute.operations.TableScanOperation;
 import com.splicemachine.derby.stream.iapi.OperationContext;
 import com.splicemachine.derby.stream.utils.StreamLogUtils;
 
-/**
- *
- *
- */
 public class SetCurrentLocatedRowAndRowKeyFunction<Op extends SpliceOperation> extends SpliceFunction<Op,ExecRow,ExecRow> {
 
+    private BaseRowLocatorFunction<Op> locatorFunction;
+
     public SetCurrentLocatedRowAndRowKeyFunction() {
         super();
+        locatorFunction = new BaseRowLocatorFunction<>();
     }
 
     public SetCurrentLocatedRowAndRowKeyFunction(OperationContext operationContext) {
         super(operationContext);
+        locatorFunction = new BaseRowLocatorFunction<>();
     }
 
     @Override
     public ExecRow call(ExecRow locatedRow) throws Exception {
         getOperation().setCurrentRow(locatedRow);
         getOperation().setCurrentRowLocation(new HBaseRowLocation(locatedRow.getKey()));
+        getOperation().setCurrentBaseRowLocation(locatorFunction.call(operationContext));
```

Comment:
We are setting this unconditionally, not just for "FOR UPDATE" statements. Is it possible to restrict it?

Reply:
You're right, I restrict it for for-update plans.